TeslaMateAgile/Services/BarryService.cs

```diff
@@ -0,0 +1,87 @@
+﻿using Microsoft.Extensions.Options;
+using System.Globalization;
+using System.Text.Json.Serialization;
+using TeslaMateAgile.Data;
+using TeslaMateAgile.Data.Options;
+using TeslaMateAgile.Services.Interfaces;
+
+namespace TeslaMateAgile.Services;
+
+public class BarryService : IPriceDataService
+{
+    private readonly HttpClient _client;
+    private readonly BarryOptions _options;
+
+    public BarryService(HttpClient client, IOptions<BarryOptions> options)
+    {
+        _client = client;
+        _options = options.Value;
+    }
+
+    public async Task<IEnumerable<Price>> GetPriceData(DateTimeOffset from, DateTimeOffset to)
+    {
+        BarryRequest request = new BarryRequest {
+            Params = new String[] { _options.BarryMPID, from.UtcDateTime.ToString("o"), to.UtcDateTime.ToString("o")}.ToList()
```

Comment:
nit: use lowercase string

```suggestion
            Params = new string[] { _options.BarryMPID, from.UtcDateTime.ToString("o"), to.UtcDateTime.ToString("o")}.ToList()
```